runtime/lua/vim/diagnostic.lua

```diff
@@ -1545,7 +1545,7 @@ function M.fromqflist(list)
   for _, item in ipairs(list) do
     if item.valid == 1 then
       local lnum = math.max(0, item.lnum - 1)
-      local col = item.col > 0 and (item.col - 1) or nil
+      local col = item.col
       local end_lnum = item.end_lnum > 0 and (item.end_lnum - 1) or lnum
       local end_col = item.end_col > 0 and (item.end_col - 1) or col
```

Comment:
I think this one needs to be changed as well.

Reply:
Hmm, I don't think so. If the qflist item doesn't contain an `end_col` then we just use the value of `col`, which would just be zero if `col` is also missing. What are you suggesting we change it to?

Reply:
you mean it is fixed by `set_diagnostic_cache` already? then no change is needed :)